de.metas.fresh/de.metas.fresh.base/src/main/jasperreports/de/metas/reports/hu_costprice/report_details.jrxml

```diff
@@ -39,9 +42,10 @@ $P{ShowDetails}
 	</variable>
 	<variable name="Product_qty" class="java.math.BigDecimal" resetType="Group" resetGroup="Product" calculation="Sum">
 		<variableExpression><![CDATA[$F{qty}]]></variableExpression>
+		<initialValueExpression><![CDATA[BigDecimal.ZERO]]></initialValueExpression>
 	</variable>
 	<variable name="Product_avg" class="java.math.BigDecimal" resetType="Group" resetGroup="Product">
-		<variableExpression><![CDATA[$V{Product_qtycost}.divide($V{Product_qty}, 2, RoundingMode.HALF_UP)]]></variableExpression>
+		<variableExpression><![CDATA[($V{Product_qty}.compareTo(BigDecimal.ZERO) == 0 ) ? BigDecimal.ZERO : $V{Product_qtycost}.divide($V{Product_qty}, 2, RoundingMode.HALF_UP)]]></variableExpression>
```

Comment:
that's OK, but for next time i recomment `<![CDATA[($V{Product_qty}.signum()==0 ) ? ...` instead..shorter :-)